beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -2051,4 +2051,149 @@ mod tests {
         assert!(connected_peers.contains(&peers[6]));
         assert!(connected_peers.contains(&peers[7]));
     }
+
+    // Test properties PeerManager should have using randomly generated input.
+    #[cfg(test)]
+    mod property_based_tests {
+        use crate::peer_manager::config::DEFAULT_TARGET_PEERS;
+        use crate::peer_manager::tests::build_peer_manager;
+        use crate::rpc::MetaData;
+        use libp2p::PeerId;
+        use quickcheck::{Arbitrary, Gen, TestResult};
+        use quickcheck_macros::quickcheck;
+        use tokio::runtime::Runtime;
+        use types::Unsigned;
+        use types::{EthSpec, MainnetEthSpec as E};
+
+        #[derive(Clone, Debug)]
+        struct PeerCondition {
+            direction: Direction,
+            attestation_subnets: Vec<AttestationSubnetId>,
+            sync_committee_subnets: Vec<SyncCommitteeSubnetId>,
+            score: f64,
+            gossipsub_score: f64,
+        }
+
+        impl Arbitrary for PeerCondition {
+            fn arbitrary(g: &mut Gen) -> Self {
+                PeerCondition {
+                    direction: Direction::arbitrary(g),
+                    attestation_subnets: Vec::arbitrary(g),
+                    sync_committee_subnets: Vec::arbitrary(g),
+                    score: f64::arbitrary(g),
+                    gossipsub_score: f64::arbitrary(g),
+                }
+            }
+        }
+
+        #[derive(Clone, Debug)]
+        enum Direction {
+            Outgoing,
+            Incoming,
+        }
+
+        impl Arbitrary for Direction {
+            fn arbitrary(g: &mut Gen) -> Self {
+                g.choose(&[Direction::Outgoing, Direction::Incoming])
+                    .unwrap()
+                    .clone()
+            }
+        }
+
+        #[derive(Clone, Debug)]
+        struct AttestationSubnetId(usize);
+
+        impl Arbitrary for AttestationSubnetId {
```

Comment:
We could probably remove these `Arbitrary` impls and just store a `bool` and some `usize`'s in `PeerCondition` right?

Then don't worry about `AttestionSubnetId`, `SyncCommitteeSubnetId` or `Direction` types